File: App.axaml.fs
namespace DLSS_5_MANAGER

open Avalonia
open Avalonia.Controls.ApplicationLifetimes
open Avalonia.Markup.Xaml
open DLSS_5_MANAGER.ViewModels
open DLSS_5_MANAGER.Views

type App() =
    inherit Application()

    override this.Initialize() =
        AvaloniaXamlLoader.Load(this)

    override this.OnFrameworkInitializationCompleted() =
        match this.ApplicationLifetime with
        | :? IClassicDesktopStyleApplicationLifetime as desktop ->
            let vm = MainViewModel()
            if not (isNull desktop.Args) then
                let args = desktop.Args
                let updateIndex = Array.tryFindIndex (fun (a: string) -> a.Equals("--updated", System.StringComparison.OrdinalIgnoreCase)) args
                if updateIndex.IsSome && updateIndex.Value + 1 < args.Length then
                    vm.ShowUpdateSuccessMessage <- true
                    vm.UpdateSuccessVersion <- args.[updateIndex.Value + 1]
            // The Screen Engine window lives on its own, so closing the main
            // window leaves it running; the app ends with the last window.
            desktop.ShutdownMode <- Avalonia.Controls.ShutdownMode.OnLastWindowClose
            let window = MainWindow(DataContext = vm)
            vm.ScreenEngine.ControlsRequested.Add(fun () ->
                Avalonia.Threading.Dispatcher.UIThread.Post(fun () -> window.ToggleScreenEngineOverlay()))
            desktop.MainWindow <- window
        | _ -> ()

        base.OnFrameworkInitializationCompleted()

File: Setup/App.axaml.cs
using Avalonia;
using Avalonia.Controls.ApplicationLifetimes;
using Avalonia.Markup.Xaml;
using Dlss5Suite.Setup.Model;
using Dlss5Suite.Setup.Views;

namespace Dlss5Suite.Setup;

public partial class App : Application
{
    /// <summary>
    /// The plan handed over by the unelevated copy, when this process is the
    /// elevated half of an install. Null on a normal run.
    /// </summary>
    public static InstallPlan? Resumed { get; set; }

    public override void Initialize() => AvaloniaXamlLoader.Load(this);

    public override void OnFrameworkInitializationCompleted()
    {
        if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
            desktop.MainWindow = new SetupWindow(Resumed);
        }

        base.OnFrameworkInitializationCompleted();
    }
}
